File: src/data_structure/linked_list/insert_a_node.cpp
/*
  Insert Node at the end of a linked list 
  head pointer input could be NULL as well for empty list
  Node is defined as 
  struct Node
  {
     int data;
     struct Node *next;
  }
*/
Node* Insert(Node *head, int data)
{
    Node *cur;
    if(head == NULL) return head;
    cur = head;
    Node *n = (struct Node *)malloc(sizeof(struct Node));
    n->data = data;
    n->next = NULL;
    while (cur->next != NULL)
    {
        cur = cur->next;
    }
    cur->next = n;
    return head;
}

File: src/data_structure/linked_list/insert_a_node.java
/*
  Insert Node at the end of a linked list 
  head pointer input could be NULL as well for empty list
  Node is defined as 
  class Node {
     int data;
     Node next;
  }
*/
Node Insert(Node head,int x) {
    if(head == null) {
        return null;
    }
    Node pre, cur, newNode;
    newNode = new Node();
    newNode.data = x;
    cur = head;
    while(cur.next != null) {
        cur = cur.next;
    }
    cur.next = newNode;
    return head;
}

